WalletWasabi.Gui/Controls/QrCode.cs

```diff
@@ -29,7 +30,37 @@ public QrCode()
 			this.WhenAnyValue(x => x.Matrix)
 				.Where(x => x != null)
 				.ObserveOn(RxApp.MainThreadScheduler)
-				.Subscribe(x => FinalMatrix = AddPaddingToMatrix(x));
+				.Do(x => FinalMatrix = AddPaddingToMatrix(x))
+				.Subscribe();
+
+			this.WhenAnyValue(x => x.QRImageSavePath)
+				.Where(x => !string.IsNullOrWhiteSpace(x) || !string.IsNullOrEmpty(x))
+				.Where(x => !(FinalMatrix is null))
+				.Subscribe(x => GenerateQRCodeBitmap(x));
+		}
+
+		private void GenerateQRCodeBitmap(string x)
+		{
+			var pixSize = PixelSize.FromSize(CoercedSize, 1);
+			using (var rtb = new RenderTargetBitmap(pixSize))
+			{
+				rtb.Render(this);
+				rtb.Save(x);
+			}
+		}
+
+		public static readonly DirectProperty<QrCode, string> QRImageSavePathProperty =
+			AvaloniaProperty.RegisterDirect<QrCode, string>(
+				nameof(QRImageSavePath),
+				o => o.QRImageSavePath,
+				(o, v) => o.QRImageSavePath = v);
+
+		private string _qRImageSavePath;
+
+		public string QRImageSavePath
+		{
+			get => _qRImageSavePath;
+			set => SetAndRaise(QRImageSavePathProperty, ref _qRImageSavePath, value);
```

Comment:
What's this strangeness? `SetAndRaise(QRImageSavePathProperty`

Reply:
@nopara73 It's a DirectProperty, regular RxUI prop are not visible to the bindings system afaik